JZD_Publish/Content/technology/owned-types.md

```diff
@@ -0,0 +1,119 @@
+
+To explain, let's look at some code:
+
+```swift
+struct MyType: Identifiable {
+    let id: Int
+    let ownerName: String
+}
+
+func printOwner(for types: [MyType], by id: MyType.ID) {
+    if let target = types.first(where: { $0.id = id }) {
+        print(target.ownerName)
+    }
+}
+```
+
+Consider the code above. There is a type that is `Identifiable`, `MyType`. That type had an `id` field that is an `Int`. Since it's `Identifiable` we can express that by `MyType.ID`, which we did in the `printOwner` function. Now, I'll show you how we can invoke that function.
+
+```swift
+let example = MyType(id: 42, ownerName: "Arthur Dent")
+
+printOwner(for types: [example], by: example.id)
+
+// Even though the way that function declares that the ID must equal MyType.ID… we don't have
+// to search by the ID that belongs to MyType. We can just use the same underlying type, 
+// which is an integer. Like so:
+
+printOwner(for types: [example], by: 42)
+printOwner(for types: [example], by: myArray.count)
+```
+
+In the above example, all of the function invocations compile fine. And that is precisely the issue we are solving for. We're aiming to reduce the prevalence of human error. That is what Phantom Types help us prevent.
+
+<br/>
+
+## Phantom Types
+
+Enter Phantom Types! Imagine, in that function declaration—where it required `MyType.ID`—that the compiler actually enforced that the function was only ever called with an `ID` that belonged to `MyType`. We can do that with Phantom Types!
+
+> We can do that with Phantom Types!
+
+We're going to build one. It will be a long an arduous journey. Buckle up! First things first… We need to define a reusable type.
+
+```swift
+struct OwnedType<Owner, Value>: RawRepresentable {
+    var rawValue: Value
+}
+```
+
+Done. That's literally it.
+
+I was joking about long and arduous btw…
+
+😂🤣
+
+It's referred to a Phantom type because the type itself doesn't actually use—at least one of—the associated type(s) that are passed into it. In this example, the Phantom Type is technically the `Owner` generic, since it's not used. These objects that are used _containing_ Phantom Types are also referred to as Phantom Types… which is exactly why I'd like to call them "Owned Types." That naming overload is rather confusing.
+
+IDK about you… but it doesn't click until I see it in action. Go ahead and modify `MyType` to use the `OwnedType` for the ID like so:
+
+```swift
+struct MyType: Identifiable {
+    let id: OwnedType<Self, Int>
+    let ownerName: String
+}
+```
+
+That's really it… Now that function above would enforce that we actually use an integer type that was associated with the `MyType`struct. Not just any old `Int`
```

Comment:
If I were reading this on the internet, I would like to see the example usage here, but that is just my own preference